server/usecases/board/project.go

```diff
@@ -0,0 +1,246 @@
+func createInitialLists(project *domain.Project, tx *sql.Tx) error {
+	closeListOption, err := FindListOptionByAction("close")
+	if err != nil {
+		return err
+	}
+	todoName := config.Element("init_list").(map[interface{}]interface{})["todo"].(string)
+	todo := list.New(
+		0,
+		project.ID,
+		project.UserID,
+		sql.NullString{String: todoName, Valid: true},
+		sql.NullString{String: "f37b1d", Valid: true},
+		sql.NullInt64{},
+		false,
+		InjectListRepository(),
+	)
+	inprogressName := config.Element("init_list").(map[interface{}]interface{})["inprogress"].(string)
+	inprogress := list.New(
+		0,
+		project.ID,
+		project.UserID,
+		sql.NullString{String: inprogressName, Valid: true},
+		sql.NullString{String: "5eb95e", Valid: true},
+		sql.NullInt64{},
+		false,
+		InjectListRepository(),
+	)
+	doneName := config.Element("init_list").(map[interface{}]interface{})["done"].(string)
+	done := list.New(
+		0,
+		project.ID,
+		project.UserID,
+		sql.NullString{String: doneName, Valid: true},
+		sql.NullString{String: "333333", Valid: true},
+		sql.NullInt64{Int64: closeListOption.ID, Valid: true},
+		false,
+		InjectListRepository(),
+	)
+	noneName := config.Element("init_list").(map[interface{}]interface{})["none"].(string)
+	none := list.New(
+		0,
+		project.ID,
+		project.UserID,
+		sql.NullString{String: noneName, Valid: true},
+		sql.NullString{String: "ffffff", Valid: true},
+		sql.NullInt64{},
+		false,
+		InjectListRepository(),
+	)
+
+	// This method only save lists.
+	// Use another methods to sync github.
+	if err := none.Create(tx); err != nil {
+		return err
+	}
+
+	if err := todo.Create(tx); err != nil {
+		return err
+	}
+	if err := inprogress.Create(tx); err != nil {
+		return err
+	}
+	if err := done.Create(tx); err != nil {
+		return err
+	}
+	return nil
+}
+
+// DeleteProject delete project and delete webhook
+func DeleteProject(projectID int64) error {
+	project, err := FindProject(projectID)
+	if err != nil {
+		return err
+	}
+
+	r, err := project.Repository(InjectRepoRepository())
+	if err == nil {
+		token, _ := project.OauthToken()
+		r.DeleteWebhook(token)
+	}
+	err = deleteLists(project)
+	if err != nil {
+		return err
+	}
+	return project.Delete()
+}
+
+// deleteLists delete all lists related a project
+func deleteLists(p *domain.Project) error {
+	lists, err := p.Lists(InjectListRepository())
+	if err != nil {
+		return err
+	}
+	for _, l := range lists {
+		err := l.DeleteTasks()
+		if err != nil {
+			return err
+		}
+		err = l.Delete()
+		if err != nil {
+			return err
+		}
+	}
+	noneList, err := p.NoneList(InjectListRepository())
+	err = noneList.DeleteTasks()
+	if err != nil {
+		return err
+	}
+	return noneList.Delete()
+}
+
+func ProjectRepository(p *domain.Project) (*repo.Repo, error) {
+	return p.Repository(InjectRepoRepository())
+}
+
+func ProjectLists(p *domain.Project) ([]*list.List, error) {
+	return p.Lists(InjectListRepository())
+}
+
+func ProjectNoneList(p *domain.Project) (*list.List, error) {
```

Comment:
__[Golint]__ exported function ProjectNoneList should have comment or be unexported ([link](https://sider.review/gh/repos/50192141/pulls/454#issue-3810554))

<sub>posted by Sider</sub>